// Copyright 2019-present Open Networking Foundation
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package gRPCUEAdmissionStatus;

service gRPCUEAdmissionStatusUpdater {
    rpc UpdateUEAdmissionStatus (gMSGUEAdmissionStatusValues) returns (gMsgResults) {}
}

message gMsgECGI {
    string plmnId = 1;
    string ecid = 2;
}

// C-RNTI, ECGI, and Status
message gMSGUEAdmissionStatusValues {
    string crnti = 1;
    gMsgECGI ecgi = 2;
    string admissionEstStatus = 3;
}

message gMsgResults {
    int32 result = 1;
}